### PR TITLE
rename whos to varinfo

### PR DESCRIPTION
**Pull request by Sacha0**
Addresses the primary 1.0 part of #12131. Best!

### CODE REVIEW

File: base/deprecated.jl [hunk 1]
```diff
@@ -1927,6 +1927,11 @@ end
     nothing
 end
 
+# deprecate whos to varinfo
+@deprecate whos(io::IO=STDOUT, m::Module=Main, pattern::Regex=r"")  varinfo(io, m, pattern)
+@deprecate whos(m::Module, pat::Regex=r"")                          varinfo(m, pat)
+@deprecate whos(pat::Regex)                                         varinfo(pat)
```
Comment (StefanKarpinski) [inline base/deprecated.jl]: Can just do `@deprecate whos varinfo`, right?

**Review by StefanKarpinski — APPROVED** [review]

File: base/deprecated.jl (hunk 1 above)
Comment (Sacha0) [reply base/deprecated.jl]: Good call! Revised accordingly. Thanks!

**Comment by JeffBezanson** [comment]
I'm working on an alternate PR to make it return a markdown table. As I see it, the options are:

1. Do nothing now, and introduce `varinfo` with better behavior later.
2. Deprecate `whos` to `varinfo` as here, but change the behavior of `varinfo` later.
3. Replace `whos` with `varinfo` that works differently now.

As far as I can tell (2) is the most breaking option, so I'll propose (3) instead.

**Comment by Sacha0** [comment]
Sounds great! Please feel welcome to close this pull request :).

**Comment by Sacha0** [comment]
Closing in favor of #24671. Thanks Jeff! :)

**Pull request closed without merging**